backend/services/core/src/datasources/Member.ts

```diff
@@ -112,4 +112,138 @@ export default class MemberAPI extends dbUtils.KnexDataSource {
+      if (!currentMember) return [];
+      /**
+       * The below query is a bit complicated, so here's a breakdown:
+       * 1. Get all pings where the current member is the sender or receiver,
+       *    and the last ping was from the other user
+       * 2. Join the members table to get the other member's info
+       * 3. Filter out the current user from the members join
+       *    (as it will return 2 both members in the ping convo)
+       * 4. Order by the last ping time, to get newest pings first
+       */
+      const unansweredPings = await this.knex
+        .select('count', 'members.*', this.knex.raw('GREATEST(??, ??) as last_ping', ['to_sent_at', 'from_sent_at']))
+        .from('pings')
+        .where((knex) =>
+          knex.where((k) =>
+            k.where('pings.from_member', '=', currentMember.id)
+              .andWhere('to_sent_at', '>', this.knex.raw('from_sent_at')))
+            .orWhere((k) =>
+              k.where('pings.to_member', '=', currentMember.id)
+                .andWhere(this.knex.raw('?? > COALESCE(??, ?)', ['from_sent_at', 'to_sent_at', 'epoch']))))
+        .join('members', (knex) => {
+          knex.on('members.id', '=', 'pings.from_member')
+            .orOn('members.id', '=', 'pings.to_member');
+        })
+        .where('members.id', '<>', currentMember.id)
+        .orderBy('last_ping', 'desc');
+
+      const mappedPings: gql.Ping[] = unansweredPings.map(({ count, last_ping, ...member }) => ({
+        counter: Math.ceil(Number.parseInt(count, 10) / 2),
+        lastPing: last_ping,
+        from: convertMember(member, ctx),
+      }));
+      return mappedPings;
+    });
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  private isTooRecent = (date: Date) =>
+    date > (new Date(Date.now() - 3_000)); // 3 seconds between pings
+
+  async canPing(ctx: context.UserContext, memberId: UUID): Promise<boolean> {
+    if (!this.hasAccess('core:member:ping', ctx)) return false;
+    const currentMember = await this.getMember(ctx, { student_id: ctx.user?.student_id });
+    if (!currentMember) return false;
+    try {
+      const memberExists = await dbUtils.unique(this.knex<sql.Member>('members').where({ id: memberId }));
+      if (!memberExists) return false;
+    } catch (e) {
+      throw new UserInputError('Member not found');
+    }
+    if (currentMember.id === memberId) return false;
+
+    // Get the ping history between the two users
+    const pingHistory = await this.knex<sql.Ping>('pings')
+      .where((knex) =>
+        knex.where('from_member', currentMember.id).andWhere('to_member', memberId))
+      .orWhere((knex) => knex.where('from_member', memberId).andWhere('to_member', currentMember.id))
+      .first();
+      // If there's no ping history or if the other user sent the most recent ping,
+      // then the current user is allowed to ping the other user.
+
+    if (!pingHistory) return true;
```

Comment:
Weird formatting of comment

```suggestion
    // If there's no ping history or if the other user sent the most recent ping,
    // then the current user is allowed to ping the other user.
    if (!pingHistory) return true;
```